File: iosApp/iosApp/Presentation/Main/Explore/ExploreUiState.swift
//
//  ExploreUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation
import Shared

struct ExploreUiState {
    var isLoading: Bool
    var isLoggedIn: Bool
    var isLoginPromptVisible: Bool
    var errorMessage: String?
    var query: String
    var selectedTab: TabType
    var selectedRegion: RegionFilter
    var selectedSort: SortFilter
    var cafes: [Cafe]
    var cafesNextCursor: String?
    var canLoadMoreCafes: Bool
    var isLoadingMoreCafes: Bool
    var maids: [Cast]
    var maidsNextCursor: String?
    var canLoadMoreMaids: Bool
    var isLoadingMoreMaids: Bool

    static let empty = ExploreUiState(
        isLoading: false,
        isLoggedIn: false,
        isLoginPromptVisible: false,
        errorMessage: nil,
        query: "",
        selectedTab: .cafe,
        selectedRegion: .all,
        selectedSort: .popular,
        cafes: [],
        cafesNextCursor: nil,
        canLoadMoreCafes: false,
        isLoadingMoreCafes: false,
        maids: [],
        maidsNextCursor: nil,
        canLoadMoreMaids: false,
        isLoadingMoreMaids: false
    )

    enum TabType: String, CaseIterable {
        case cafe = "카페"
        case maid = "캐스트"
    }

    enum RegionFilter: String, CaseIterable {
        case all = "all"
        case seoul = "seoul"
        case tokyo = "tokyo"
        case osaka = "osaka"

        var label: String {
            switch self {
            case .all: return "전체"
            case .seoul: return "서울"
            case .tokyo: return "도쿄"
            case .osaka: return "오사카"
            }
        }
    }

    enum SortFilter: String, CaseIterable {
        case popular = "popular"
        case latest = "latest"
        case rating = "rating"

        var label: String {
            switch self {
            case .popular: return "인기순"
            case .latest: return "최신순"
            case .rating: return "평점순"
            }
        }
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/explore/ExploreUiState.kt
package com.hhp227.concafe.presentation.main.explore

import com.hhp227.concafe.domain.model.Cafe
import com.hhp227.concafe.domain.model.Cast

data class ExploreUiState(
    val isLoading: Boolean = false,
    val isLoggedIn: Boolean = false,
    val isLoginPromptVisible: Boolean = false,
    val errorMessage: String? = null,
    val query: String = "",
    val selectedTab: TabType = TabType.CAFE,
    val selectedRegion: RegionFilter = RegionFilter.ALL,
    val selectedSort: SortFilter = SortFilter.POPULAR,
    val cafes: List<Cafe> = emptyList(),
    val hasLoadedCafes: Boolean = false,
    val cafesNextCursor: String? = null,
    val canLoadMoreCafes: Boolean = false,
    val isLoadingMoreCafes: Boolean = false,
    val maids: List<Cast> = emptyList(),
    val hasLoadedMaids: Boolean = false,
    val maidsNextCursor: String? = null,
    val canLoadMoreMaids: Boolean = false,
    val isLoadingMoreMaids: Boolean = false
) {
    enum class TabType(val label: String) {
        CAFE("카페"),
        MAID("캐스트")
    }

    enum class RegionFilter(val label: String, val key: String) {
        ALL("전체", "all"),
        SEOUL("서울", "seoul"),
        BUSAN("부산", "busan"),
        DAEGU("대구", "daegu"),
        TOKYO("도쿄", "tokyo"),
        OSAKA("오사카", "osaka"),
        ETC("그 외지역", "etc")
    }

    enum class SortFilter(val label: String, val key: String) {
        POPULAR("인기순", "popular"),
        LATEST("최신순", "latest"),
        RATING("평점순", "rating")
    }

    companion object {
        fun empty() = ExploreUiState()
    }
}
